### ISSUE TITLE
X coordinates collision

### ISSUE THREAD

**Issue by gching** [SSSaaS/sssa-ruby#1]
https://github.com/SSSaaS/sssa-ruby/blob/df65db5785d591630e4acdff7d2a65b52d2d9581/lib/sssa.rb#L31

I know this codebase is pretty old, but it seems like there is a possibility of x coordinates colliding upon generating the points / parts at random.

As we do not insert a selected x coordinate for a point into `numbers`, another point's x coordinate can possibly collide.

**Comment by cipherboy** [comment]
Fair point, I do believe that's an oversight on our part. I probably won't fix it though -- the probability of it occurring is extraordinarily low considering the prime in question (`115792089237316195423570985008687907853269984665640564039457584007913129639747`) and a reasonable quantity of shares (< 2^16). 

If you want to file a PR, fine, but neither I or @Mab879 are currently using this. I'd also suggest the protocol should be changed before others use this, so I'm not sure its worth your time.

**Comment by gching** [comment]
Makes sense 👍 was just researching up a lot of sss algorithms. Great clean code btw!
